### PR TITLE
Implement AI for Arena

### PR DESCRIPTION
**Pull request by Agetian**

### CODE REVIEW

File: forge-gui/res/cardsfolder/a/arena.txt [hunk 1]
```diff
@@ -1,9 +1,8 @@
 Name:Arena
 ManaCost:no cost
 Types:Land
-A:AB$ Tap | Cost$ 3 T | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Select target creature you control | SubAbility$ DBTap | AlwaysRemember$ True | SpellDescription$ Tap target creature you control and target creature of an opponent's choice they control. Those creatures fight each other. (Each deals damage equal to its power to the other.)
+A:AB$ Tap | Cost$ 3 T | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Select target creature you control | SubAbility$ DBTap | AlwaysRemember$ True | AILogic$ Arena | SpellDescription$ Tap target creature you control and target creature of an opponent's choice they control. Those creatures fight each other. (Each deals damage equal to its power to the other.)
 SVar:DBTap:DB$ Tap | TargetingPlayer$ Player.Opponent | TargetingPlayerControls$ True | ValidTgts$ Creature | TgtPrompt$ Select target creature you control | SubAbility$ DBFight | AlwaysRemember$ True
 SVar:DBFight:DB$ Fight | Defined$ Remembered | SubAbility$ DBCleanup
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/a/arena.txt:6]: imo `Defined$ Targeted` should work

File: forge-gui/res/cardsfolder/a/arena.txt [hunk 2]
```diff
@@ -1,9 +1,8 @@
 Name:Arena
 ManaCost:no cost
 Types:Land
-A:AB$ Tap | Cost$ 3 T | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Select target creature you control | SubAbility$ DBTap | AlwaysRemember$ True | SpellDescription$ Tap target creature you control and target creature of an opponent's choice they control. Those creatures fight each other. (Each deals damage equal to its power to the other.)
+A:AB$ Tap | Cost$ 3 T | ValidTgts$ Creature.YouCtrl | TgtPrompt$ Select target creature you control | SubAbility$ DBTap | AlwaysRemember$ True | AILogic$ Arena | SpellDescription$ Tap target creature you control and target creature of an opponent's choice they control. Those creatures fight each other. (Each deals damage equal to its power to the other.)
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/a/arena.txt:4]: _Magus of the Arena_ should get same treatment ;)

File: forge-gui/res/cardsfolder/a/arena.txt (hunk 1 above)
Comment (Agetian) [reply forge-gui/res/cardsfolder/a/arena.txt]: Done :) Tested it and it does seem to work fine indeed.

File: forge-gui/res/cardsfolder/a/arena.txt (hunk 2 above)
Comment (Agetian) [reply forge-gui/res/cardsfolder/a/arena.txt]: Hehe, indeed, thanks! :) Done

**Review by tool4ever — APPROVED** [review]

**Pull request merged**